ompi/mca/coll/base/coll_base_allreduce.c

```diff
@@ -652,12 +661,14 @@ ompi_coll_base_allreduce_intra_ring_segmented(const void *sbuf, void *rbuf, int
     segcount = count;
     COLL_BASE_COMPUTED_SEGCOUNT(segsize, typelng, segcount)
 
-        /* Special case for count less than size * segcount - use regular ring */
-        if (count < (size * segcount)) {
-            OPAL_OUTPUT((ompi_coll_base_framework.framework_output, "coll:base:allreduce_ring_segmented rank %d/%d, count %d, switching to regular ring", rank, size, count));
-            return (ompi_coll_base_allreduce_intra_ring(sbuf, rbuf, count, dtype, op,
-                                                         comm, module));
-        }
+    /* Special case for count less than size * segcount - use regular ring */
```

Comment:
Why is this different than above ?

Reply:
The size test is different, the rest should be the same?